cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,19 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = null;
+      if (listeners != null && listeners.length() != 0) {
+        String firstListener = listeners.split("\\s*,\\s*")[0];
+        String[] allListeners = firstListener.split(":");
+        if (allListeners[1].length() == 2) {
+          bootstrapServers = "0.0.0.0:" + allListeners[2];
```

Comment:
Why do we switch to `0.0.0.0` from `localhost`?

Reply:
> Why do we switch to `0.0.0.0` from `localhost`?
such as SSL://:9091，host is empty, new EndPoint  host will be assigned null
' if (host == null || host.trim.isEmpty)' 
'anyLocalAddress.holder().hostName = "0.0.0.0" '

Reply:
l misunderstood
such as SSL://:9091,kafka server will listen to 0.0.0.0, but as kafka client we should use localhost